packages/dev/core/src/Culling/ray.ts

```diff
@@ -705,26 +705,24 @@ Scene.prototype.createPickingRayToRef = function (
 ): Scene {
     const engine = this.getEngine();
 
-    if (!camera) {
-        if (!this.activeCamera) {
-            return this;
-        }
-
-        camera = this.activeCamera;
+    if (!camera && !(camera = this.activeCamera!)) {
+        return this;
     }
 
     const cameraViewport = camera.viewport;
-    const viewport = cameraViewport.toGlobal(engine.getRenderWidth(), engine.getRenderHeight());
+    const renderHeight = engine.getRenderHeight();
+    const { x: vx, y: vy, width, height } = cameraViewport.toGlobal(engine.getRenderWidth(), renderHeight);
 
     // Moving coordinates to local viewport world
-    x = x / engine.getHardwareScalingLevel() - viewport.x;
-    y = y / engine.getHardwareScalingLevel() - (engine.getRenderHeight() - viewport.y - viewport.height);
+    const levelInv = 1 / engine.getHardwareScalingLevel();
+    x = x * levelInv - vx;
+    y = y * levelInv - (renderHeight - vy - height);

```

Comment:
that should fail, if x and y are consts. Same goes to the next function

Also note - the reason we don't use spread too much is that in UMD it brings no benefits, as it is not a feature of es5.

Reply:
and if it doesn't we need to check why exactly.

Reply:
`Viewport.x` is remapped to `vx`, not to `x` - `x` is a parameter to the method.

Reply:
Yep, but I can't see the definition of x and y.  Unless I am missing something? (which I might be, of course :-) )